src/soundio/sounddeviceportaudio.cpp

```diff
@@ -107,8 +107,7 @@ SoundDevicePortAudio::SoundDevicePortAudio(UserSettingsPointer config,
     // Setting parent class members:
     m_hostAPI = Pa_GetHostApiInfo(deviceInfo->hostApi)->name;
     m_dSampleRate = deviceInfo->defaultSampleRate;
-    m_strInternalName = QString("%1, %2").arg(QString::number(m_devId),
-            deviceInfo->name);
+    m_strInternalName = deviceInfo->name;
```

Comment:
Even now I often have to reconfigure my settings depending on which devices are found. I don't think that this fix causes any severe usability issues other than that we already have.

Reply:
This is a Linux only issue, where devices in use are not enumerated. 
After merging this PR, all users have to reconfigure their Hardware preferences. 
This is a bad idea since the minority of our users are on Windows.

Reply:
No. You are wrong.
This has been happening on Windows too.
That index changes, sometimes every reboot.
I assume that with "backwards compatibility" you mean when loading on an older Mixxx the configuration created with this PR.
 That might be worth considering for development and people trying before upgrading. In fact, keeping compatibility could beneficiate the newer version more than the older one, in this case. So yes, trying to ignore the index on load rather than not saving it could be a better solution, and documenting why.
Finally, there is still that problem that you mention wih the name and "hw:xx". What do you propose?

Reply:
I mean: on Linux the issue is worse, because it also happens without reboot, if a device is in use by an other application (pulse). 

> Finally, there is still that problem that you mention wih the name and "hw:xx". What do you propose?

We need a special case that excludes the hw:x,x and the index if not found. How about looking for the full string fist and if not found use the inner part only.

Reply:
Mixxx users are accustomed to having to reconfigure their sound hardware frequently, so I am not concerned about making them do it once more.

Reply:
I am very certain nobody minds to reconfigure their audio config. That's probably the task which all users do regularly (switching equipment) and it's easy.